packages/trader/src/Modules/Trading/Components/Elements/mobile-widget.tsx

```diff
@@ -0,0 +1,127 @@
+import { useTraderStore } from 'Stores/useTraderStores';
+
+const MobileWidget = observer(
+    ({ toggleDigitsWidget, is_collapsed }: { toggleDigitsWidget: () => void; is_collapsed: boolean }) => {
+        const { ui } = useStore();
+        const { onChangeUiStore } = ui;
+        const trade_store = useTraderStore();
+        const {
+            amount,
+            basis,
+            currency,
+            duration,
+            duration_min_max,
+            duration_unit,
+            form_components,
+            is_multiplier,
+            last_digit,
+            onChange,
+        } = trade_store;
+
+        const [is_open, setIsOpen] = React.useState(false);
+        const [tab_index, setTabIndex] = React.useState(0);
+
+        React.useEffect(() => {
+            assertDurationIsWithinBoundary();
+        });
+
+        const assertDurationIsWithinBoundary = () => {
+            const contract_expiry_type = getExpiryType(trade_store);
+            const [min_value, max_value] = getDurationMinMaxValues(
+                duration_min_max,
+                contract_expiry_type,
+                duration_unit
+            );
+
+            if (contract_expiry_type === 'tick' && duration < Number(min_value)) {
+                onChangeUiStore({ name: `duration_${duration_unit}`, value: min_value });
+                onChange({ target: { name: 'duration', value: min_value } });
+            }
+            if (duration >= Number(min_value) && duration > Number(max_value)) {
+                onChangeUiStore({ name: `duration_${duration_unit}`, value: max_value });
+                onChange({ target: { name: 'duration', value: max_value } });
+            }
+        };
+
+        // tab_index 0 is duration and 1 is amount
+        const toggleWidget = (tab_index: number) => {
+            setTabIndex(tab_index);
+            setIsOpen(!is_open);
+        };
+
+        const getHumanReadableDuration = () => {
+            if (!duration_unit) return '';
+            const lookup = {
+                t: [<Localize i18n_default_text='tick' key='t' />, <Localize i18n_default_text='ticks' key='ts' />],
+                s: [<Localize i18n_default_text='second' key='s' />, <Localize i18n_default_text='seconds' key='ss' />],
+                m: [<Localize i18n_default_text='min' key='m' />, <Localize i18n_default_text='mins' key='ms' />],
+                h: [<Localize i18n_default_text='hour' key='h' />, <Localize i18n_default_text='hours' key='hs' />],
+                d: [<Localize i18n_default_text='day' key='d' />, <Localize i18n_default_text='days' key='ds' />],
+            };
+
+            try {
+                const formatted_duration_unit =
+                    +duration === 1
+                        ? lookup[duration_unit as keyof typeof lookup][0]
+                        : lookup[duration_unit as keyof typeof lookup][1];
+                return [duration, ' ', formatted_duration_unit];
```

Comment:
let's leave it as it was, with the `if` condition:
 ![image](https://github.com/binary-com/deriv-app/assets/103177211/0bd16626-4263-4489-9fa5-9b899c046609)
and add a condition to the return statement:
`return duration && formatted_duration_unit ? `${duration} ${formatted_duration_unit}` : '';`